test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_cad.cxx

```diff
@@ -365,6 +367,185 @@ TEST (t8_gtest_geometry_cad, jacobian)
 }
 #endif /* T8_WITH_OCC */
 
+#if T8_WITH_OCC
+/* The test checks if the mapping algorithms for curved 2d elements do not shift values on an edge which is not curved.
+ * In that case, the cad geometry should output the same out_coords as the linear geometry function. */
+class class_2d_element_linear_cad_curve: public testing::TestWithParam<t8_eclass_t> {
+ protected:
+  void
+  SetUp () override
+  {
+    eclass = GetParam ();
+    T8_ASSERT (0 <= eclass && eclass < T8_ECLASS_COUNT);
+    Handle_Geom_Curve cad_curve;
+    TColgp_Array1OfPnt point_array (1, 3);
+
+    /*  x--> u-parameter
+    *   
+    *                 curve
+    *  ----------------------------------
+    * 
+    *  0               0.5              1
+    */
+
+    point_array (1) = gp_Pnt (0.0, 0.0, 0.0);
+    point_array (2) = gp_Pnt (0.5, 0.0, 0.0);
+    point_array (3) = gp_Pnt (1.0, 0.0, 0.0);
+
+    cad_curve = GeomAPI_PointsToBSpline (point_array).Curve ();
+    shape = BRepBuilderAPI_MakeEdge (cad_curve).Edge ();
+  }
+
+  void
+  TearDown () override
+  {
+    /* The cmesh is destroyed in the test itself. */
+  }
+  t8_cmesh_t cmesh;
+  t8_eclass_t eclass;
+  TopoDS_Shape shape;
+  /* The arrays prescribe the linkage of the element. Edge 2 of the element is linked and the face is unlinked. */
+  int faces[1] = { 0 };
+  int edges[8] = { 0, 0, 1, 0, 0, 0, 0, 0 };
+
+  const double test_ref_coords[9] = { 0.0, 0.0, 0.0, 0.5, 0.0, 0.0, 1.0, 0.0, 0.0 };
+  /* TODO: use randomised test_ref_coords (should lie on linked curve), because the out_coords should be the same, no matter the test_ref_coord. */
+};
+
+TEST_P (class_2d_element_linear_cad_curve, t8_check_2d_element_linear_cad_curve)
+{
+  /* Saving the corner vertices for the given element class. */
+  const int num_vertices = t8_eclass_num_vertices[eclass];
+  const double vertices_tri[27] = { 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0,
+                                    0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0 };
+  const double vertices_quad[48] = { 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0,
+                                     1.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0,
+                                     0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0 };
+  /* Edges are parametrized in one parameter u. The array contains the parameters
+   * each vertex of the edge has on the linked curve. */
+  double params[2] = { 0, 1 };
+
+  for (size_t i_orientation = 0; i_orientation < (eclass == T8_ECLASS_QUAD ? 4 : 3); ++i_orientation) {
+    const int orientation = i_orientation * (eclass == T8_ECLASS_QUAD ? 12 : 9);
```

Comment:
```suggestion
    const int orientation = i_orientation * t8_eclass_num_vertices[eclass] * T8_ECLASS_MAX_DIM;
```
We should avoid using magic numbers and use the already defined variables for that.